client/app/router.coffee

```diff
@@ -1,96 +1,156 @@
-            fluxAction: 'showComposeMessage'
-
-        'default':
-            pattern: ''
-            fluxAction: ''
-
-    # default route
-    routes: '': 'default'
-
-    # Determines and gets the default parameters regarding a specific action
-    _getDefaultParameters: (action, parameters) ->
-        switch action
-
-            when 'account.mailbox.messages'
-                defaultAccountID = AccountStore.getDefault()?.get 'id'
-                # if parameters contains accountID but no mailboxID,
-                # get the default mailbox for this account
-                if parameters.accountID?
-                    mailbox = AccountStore.getDefaultMailbox parameters.accountID
-                else
-                    mailbox = AccountStore.getDefaultMailbox defaultAccountID
-                defaultMailboxID = mailbox?.get 'id'
-                defaultParameters = {}
-                defaultParameters.accountID = defaultAccountID
-                defaultParameters.mailboxID = defaultMailboxID
-                defaultParameters.pageAfter = '-'
-                defaultParameters.sort = '-date'
-                defaultParameters.after = '-'
-                defaultParameters.before = '-'
-                defaultParameters.type = 'nofilter'
-                defaultParameters.flag = '-'
-
-            when 'account.config'
-                defaultAccount = AccountStore.getDefault()?.get 'id'
-                defaultParameters =
-                    accountID: defaultAccount
-                    tab: 'account'
-
-            when 'search'
-                defaultParameters =
-                    accountID: 'all'
-                    search: '-'
-
-            else
-                defaultParameters = null
-
-        return defaultParameters
+
+LayoutActionCreator = require './actions/layout_action_creator'
+
+RouteGetter = require './getters/route'
+
+{ActionTypes} = require './constants/app_constants'
+
+_ = require 'lodash'
+
+PREFIX_ACCOUNT = 'account/:accountID/mailbox/:mailboxID'
+
+# MessageList :
+# ?sort=asc&filters=&status=unseen&start=2016-02-27T23:00:00.000Z&end=2016-03-05T22:59:59.999Z
+
+# Search :
+# #account/3510d24990c596125ecc9e1fc800616a/mailbox/3510d24990c596125ecc9e1fc80064d3/search/?q=plop
+
+__routes = {}
+
+class Router extends Backbone.Router
+
+    routes:
+        '/*'                                    : 'messageList'
+        'account/new'                           : 'accountEdit'
+        'account/:accountID/config/:tab'        : 'accountNew'
+        'search/?q=:search'                     : 'search'
+        ':messageID'                            : 'messageShow'
+        ':messageID/edit'                       : 'messageEdit'
+        'new'                                   : 'messageNew'
+        ':messageID/forward'                    : 'messageForward'
+        ':messageID/reply'                      : 'messageReply'
+        ':messageID/reply-all'                  : 'messageReplyAll'
```

Comment:
I'm not sure about those routes: is your goals to mimic the https://github.com/cozy/cozy-emails/pull/783/files#diff-f43c9098b0e1615b73bef5226fbe582f definition at final form, or is this just transitional?